pkg/cluster/memberlist.go

```diff
@@ -0,0 +1,54 @@
+package cluster
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"strconv"
+	"time"
+
+	"github.com/hashicorp/memberlist"
+)
+
+type delegate interface {
+	memberlist.EventDelegate
+	memberlist.Delegate
+}
+
+func NewMemberlist(name, addr string, secret []byte, verbose bool, d delegate) (*memberlist.Memberlist, error) {
+	if len(secret) != 32 {
```

Comment:
in a follow up we need to put magic numbers into consts

Reply:
:crystal_ball: :+1: